docs/callbacks.rst

```diff
@@ -0,0 +1,119 @@
+.. role:: python(code)
+          :language: python
+
+.. _callbacks:
+============================================================
+Callbacks
+============================================================
+
+LBANN has numerous callbacks that can be used to collect
+data about an experiment, such as scalars, metrics, weights,
+memory usage, images, etc. See the :ref:`Available
+Callbacks<available-callbacks>` section for a comprehensive list.
+
+The callbacks are set to execute at various times, and can be
+used for reporting, debugging, and performing advanced training
+techniques.
+
+For a complete listing of callbacks and details about their
+functionality, Please see :ref:`Available
+Callbacks<available-callbacks>`.
+
+.. _using-callbacks:
+------------------------------------------------
+Using Callbacks
+------------------------------------------------
+
+Callbacks are used by adding them to the python front end with the
+appropriate arguments and passing them as a list into the model.
+For example, the callbacks timer, print_statistics, and save model
+could be included with the following:
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Python Front End
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: python
```

Comment:
For both the PFE and the proto examples, I would put in the sample layer definitions before referring to the names so that it's clear that, e.g., "top1" refers to a categorical accuracy layer.